// Copyright 2023 Intrinsic Innovation LLC

syntax = "proto3";

package intrinsic_proto.geometry;

import "intrinsic/geometry/service/geometry_storage_refs.proto";
import "intrinsic/math/proto/matrix.proto";

// A proto holding the Geometry refs and a transform similar to the
// TransformedGeometry object.
message TransformedGeometryStorageRefs {
  // The references to the stored geometry.
  GeometryStorageRefs geometry_storage_refs = 1;

  // This is a 4x4 affine transform from the reference frame to this shape.
  intrinsic_proto.Matrixd ref_t_shape_aff = 2;
}
